### DISCUSSION TITLE
another problem with controls and arrays

### DISCUSSION THREAD

**Discussion by Drugo67** [VBAndCs/sVB-Small-Visual-Basic#91] [Q&A]
Hello everyone! I had already asked a question regarding control arrays and in fact I was partially answered but now I have another problem. Being used to the good old VB6 I often used arrays of controls. But with sVB this is not possible, so if I have for example 25 labels on the form (label1, label2... label25) and I had to change the text of only one of them depending on a number (such as math.GetRandomNumber(25)) is there a possibility to do it without doing 25 "If Then" loops? That is, how can I identify a certain label depending on a result? Thanks in Advance
Drugo

**Comment by Drugo67** [comment]
based on the answer Mohammad gave me earlier, and assuming I have 5 labels (label1, label2... label5) identified with a certain number at the end of the name, I made this little routine to write to a certain label based on a result. Is there a better or more elegant way to access and write to it?

num = Math.GetRandomNumber(5)
   
     ForEach _label In Me.Controls
      If _label.Name = "Label" & Text.ToStr(num) Then
         _label.Text = "ok label - " & Lblnum.Text
      EndIf
   Next

**Comment by VBAndCs** [comment]
For short, you can use:
`If _Label.Name = "Label" & Num Then`
as sVB makes the necessary type conversions.
But here is another version of the code that doesn't require a special naming pattern for labels:
Num = Math.GetRandomNumber(5)
N = 0  ' represents the found label count
ForEach _Label In Me.Controls
   If _Label.TypeName = ControlTypes.Label Then
      N = N + 1
      If N = Num Then
         _Label.Text = "ok label - " & Lblnum.Text
         _Label.FitContentWidth()
      EndIf
   EndIf
Next



________________________________

**Comment by Drugo67** [comment]
Thank You Mohammad!!! Good to know!
